arm/compute/client.go

```diff
@@ -0,0 +1,2031 @@
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator 0.11.0.0
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package compute
+
+import (
+	"fmt"
+	"github.com/azure/go-autorest/autorest"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const (
+	ApiVersion             = "2015-06-15"
+	DefaultBaseUri         = "https://management.azure.com"
+	DefaultPollingDuration = 10 * time.Minute
+)
+
+type ComputeManagementClient struct {
+	autorest.Client
+	BaseUri        string
+	SubscriptionId string
+}
+
+func New(subscriptionId string) *ComputeManagementClient {
+	return NewWithBaseUri(DefaultBaseUri, subscriptionId)
+}
+
+func NewWithBaseUri(baseUri string, subscriptionId string) *ComputeManagementClient {
+	client := &ComputeManagementClient{BaseUri: baseUri, SubscriptionId: subscriptionId}
+	client.PollingMode = autorest.PollUntilDuration
+	client.PollingDuration = DefaultPollingDuration
+	return client
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// AvailabilitySets Client
+//
+////////////////////////////////////////////////////////////////////////////////
+type AvailabilitySetsClient struct {
+	ComputeManagementClient
+}
+
+func NewAvailabilitySetsClient(subscriptionId string) *AvailabilitySetsClient {
+	return NewAvailabilitySetsClientWithBaseUri(DefaultBaseUri, subscriptionId)
+}
+
+func NewAvailabilitySetsClientWithBaseUri(baseUri string, subscriptionId string) *AvailabilitySetsClient {
+	return &AvailabilitySetsClient{*NewWithBaseUri(baseUri, subscriptionId)}
+}
+
+// Delete the operation to delete the availability set.;
+//
+// resourceGroupName is the name of the resource group. availabilitySetName is
+// the name of the availability set.
+func (client *AvailabilitySetsClient) Delete(resourceGroupName string, availabilitySetName string) (err error) {
+
+	req, err := client.NewDeleteRequest(resourceGroupName, availabilitySetName)
+	if err != nil {
+		return fmt.Errorf("compute: Failure preparing AvailabilitySetsClient.Delete request (%v)", err)
+	}
+
+	req, err = autorest.Prepare(
+		req,
+		client.WithAuthorization(),
+		client.WithInspection())
+	if err != nil {
+		return fmt.Errorf("compute: Failure sending AvailabilitySetsClient.Delete request (%v)", err)
+	}
+
+	resp, err := autorest.SendWithSender(client, req)
+	if err != nil {
+		return fmt.Errorf("compute: Failure sending AvailabilitySetsClient.Delete request (%v)", err)
+	}
+
+	err = autorest.Respond(
+		resp,
+		client.ByInspecting(),
+		autorest.WithErrorUnlessOK(),
+		autorest.ByClosing())
+
+	return
+}
+
+func (client *AvailabilitySetsClient) NewDeleteRequest(resourceGroupName string, availabilitySetName string) (*http.Request, error) {
+
+	pathParameters := map[string]interface{}{
+		"availabilitySetName": url.QueryEscape(availabilitySetName),
```

Comment:
@ahmetalpbalkan URL encoding is *not* idempotent. That is, if a parameter is already encoded, encoding again does "bad things." It is a common problem. The x-ms-skip-url-encoding switch gives the Swagger author the ability to control. No harm, no foul.